tests/drivers/adc/adc_api/boards/nucleo_u083rc.overlay

```diff
@@ -0,0 +1,36 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright (c) 2024 STMicroelectronics
+ */
+
+/ {
+	zephyr,user {
+		/* adjust channel number according to pinmux in board.dts */
+		io-channels = <&adc1 0>, <&adc1 1>;
+	};
+};
+
+&adc1 {
+	st,adc-clock-source = <ASYNC>;
+	clocks = <&rcc STM32_CLOCK_BUS_APB1_2 0x00100000>,
+		 <&rcc STM32_SRC_HSI ADC_SEL(0)>;
```

Comment:
If you want to select HSI, this should be `ADC_SEL(2)`. `ADC_SEL(0)` will select the System Clock.

Besides, this clock configuration should be done directly in the board dts.

Reply:
Indeed, 
I did the tests with your suggestion,I don't know how I managed not to add these configurations after . 
Thank you.